Source/Scene/VertexAttributeSemantic.js

```diff
@@ -0,0 +1,274 @@
+    case VertexAttributeSemantic.WEIGHTS:
+    case VertexAttributeSemantic.FEATURE_ID:
+      return true;
+    //>>includeStart('debug', pragmas.debug);
+    default:
+      throw new DeveloperError("semantic is not a valid value.");
+    //>>includeEnd('debug');
+  }
+};
+
+/**
+ * Gets the vertex attribute semantic matching the glTF semantic.
+ *
+ * @param {String} gltfSemantic The glTF semantic.
+ *
+ * @returns {VertexAttributeSemantic|undefined} The vertex attribute semantic, or undefined if there is no match.
+ *
+ * @private
+ */
+VertexAttributeSemantic.fromGltfSemantic = function (gltfSemantic) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.string("gltfSemantic", gltfSemantic);
+  //>>includeEnd('debug');
+
+  var semantic = gltfSemantic;
+
+  // Strip the set index from the semantic
+  var setIndexRegex = /^(\w+)_\d+$/;
+  var setIndexMatch = setIndexRegex.exec(gltfSemantic);
+  if (setIndexMatch !== null) {
+    semantic = setIndexMatch[1];
+  }
+
+  switch (semantic) {
+    case "POSITION":
+      return VertexAttributeSemantic.POSITION;
+    case "NORMAL":
+      return VertexAttributeSemantic.NORMAL;
+    case "TANGENT":
+      return VertexAttributeSemantic.TANGENT;
+    case "TEXCOORD":
+      return VertexAttributeSemantic.TEXCOORD;
+    case "COLOR":
+      return VertexAttributeSemantic.COLOR;
+    case "JOINTS":
+      return VertexAttributeSemantic.JOINTS;
+    case "WEIGHTS":
+      return VertexAttributeSemantic.WEIGHTS;
+    case "_FEATURE_ID":
+    case "_BATCHID": // for b3dm compatibility
+    case "BATCHID": // for legacy b3dm compatibility
+      return VertexAttributeSemantic.FEATURE_ID;
+  }
+
+  return undefined;
+};
+
+/**
+ * Gets the vertex attribute semantic matching the pnts semantic.
+ *
+ * @param {String} pntsSemantic The pnts semantic.
+ *
+ * @returns {VertexAttributeSemantic|undefined} The vertex attribute semantic, or undefined if there is no match.
+ *
+ * @private
+ */
+VertexAttributeSemantic.fromPntsSemantic = function (pntsSemantic) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.string("pntsSemantic", pntsSemantic);
+  //>>includeEnd('debug');
+
+  switch (pntsSemantic) {
+    case "POSITION":
+    case "POSITION_QUANTIZED":
+      return VertexAttributeSemantic.POSITION;
+    case "RGBA":
+    case "RGB":
+    case "RGB565":
+      return VertexAttributeSemantic.COLOR;
+    case "NORMAL":
+    case "NORMAL_OCT16P":
+      return VertexAttributeSemantic.NORMAL;
+    case "BATCH_ID":
+      return VertexAttributeSemantic.FEATURE_ID;
+    //>>includeStart('debug', pragmas.debug);
+    default:
+      throw new DeveloperError("pntsSemantic is not a valid value.");
+    //>>includeEnd('debug');
+  }
+};
+
+/**
+ * Gets the shader type for the given vertex attribute.
+ *
+ * @param {VertexAttributeSemantic} semantic The semantic.
+ *
+ * @returns {String} The shader type.
+ *
+ * @private
+ */
+VertexAttributeSemantic.getShaderType = function (semantic) {
```

Comment:
I think `getGlslType` is a better function name. `ShaderType` seems to reference the type of shader, for example: https://github.com/CesiumGS/cesium/blob/c2e3e401f9f4227d60a8ba3ca568200ed69546cb/Specs/getWebGLStub.js#L277

Reply:
I flip flopped about this myself and I'm ok with this suggestion.

Heads up that some variable names might have to be changed in `model-loading`.

Reply:
I also like `GlslType`, this will be more consistent with some things for metadata packing in the future